plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -283,10 +295,16 @@ private static ConnectorPageSource createOrcPageSource(
                     .setBloomFiltersEnabled(options.isBloomFiltersEnabled());
             Map<IcebergColumnHandle, Domain> effectivePredicateDomains = effectivePredicate.getDomains()
                     .orElseThrow(() -> new IllegalArgumentException("Effective predicate is none"));
+
+            Optional<ReaderColumns> columnProjections = projectColumns(columns);
+
             List<OrcColumn> fileReadColumns = new ArrayList<>(columns.size());
             List<Type> fileReadTypes = new ArrayList<>(columns.size());
             List<ColumnAdaptation> columnAdaptations = new ArrayList<>(columns.size());
-            for (IcebergColumnHandle column : columns) {
+            List<IcebergColumnHandle> readColumns = columnProjections
+                    .map(readerColumns -> readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toList()))
+                    .orElse(columns);
+            for (IcebergColumnHandle column : readColumns) {
                 OrcColumn orcColumn;
                 if (fileColumnsByIcebergId.isEmpty()) {
```

Comment:
Is that possible at this point?

Reply:
It looks like it builds a backup mapping by name for the columns where the id is empty